tests/v2/validation/provisioning/k3s/custom_cluster_test.go

```diff
@@ -159,6 +163,15 @@ func (c *CustomClusterProvisioningTestSuite) ProvisioningK3SCustomCluster(extern
 				clusterToken, err := clusters.CheckServiceAccountTokenSecret(client, clusterName)
 				require.NoError(c.T(), err)
 				assert.NotEmpty(c.T(), clusterToken)
+
+				err = hardening.CheckClusterHardening(client, tt.hardening.Hardened, nodes, tt.nodeRoles)
+				require.NoError(c.T(), err)
+
+				hardenCluster := clusters.HardenK3SRKE2ClusterConfig(clusterName, namespace, "", "", kubeVersion, nil)
+
+				hardenClusterResp, err := clusters.UpdateK3SRKE2Cluster(client, clusterResp, hardenCluster)
```

Comment:
So basically if hardened is set to false, you will still go through these checks updates? Why don't you check for hardening at the test level, and then check and update accordingly. If it's not hardened.

Reply:
@slickwarren and I were talking of this as well, yeah I'll definitely look into that. Cause although it works for now, there isn't really an explanation behind it. Not a logical one at least lol...

Reply:
Yeah for sure, like why go through these checks and updates if hardened is set to false. Thank you!